lib/smart_app_launch/token_introspection.rb

```diff
@@ -0,0 +1,227 @@
+require_relative 'token_exchange_test'
+require_relative 'token_refresh_body_test'
+
+module SMARTAppLaunch
+  class TokenIntrospectionTestGroup < Inferno::TestGroup
+    title 'Token Introspection Group'
+    description %(
+      # Background
+      OAuth 2.0 Token introspection, as described in [RFC-7662](https://datatracker.ietf.org/doc/html/rfc7662), allows
+      an authorized resource server to query an OAuth 2.0 authorization server for metadata on a token.  The
+      [SMART App Launch STU 2.1 Implementation Guide Section on Token Introspection](https://hl7.org/fhir/smart-app-launch/token-introspection.html)
+      states that "SMART on FHIR EHRs SHOULD support token introspection, which allows a broader ecosystem of resource servers
+      to leverage authorization decisions managed by a single authorization server."
+
+      # Test Methodology
+      For these tests, Inferno acts as an authorized resource server that queries the authorization server about an access 
+      token, rather than a client to a FHIR resource server as in the previous SMART App Launch tests.  The tests will 
+      create a request to the authorization server's token introspection endpoint and validate the introspection response.
+
+      The means of discovery of the token introspection endpoint are outside the scope of the RFC-7662 specification.
+      As such, Inferno makes no assumptions that that the introspection endpoint is included in the `.well-known` endpoint query and leaves it to be input by the user. 
+      
+      To complete the tests, Inferno should be registered with the authorization server as an authorized resource server
+      capable of accessing the token introspection endpoint.  RFC-7662 requires "some form of authorization" to access
+      the token endpoint but does specifiy any one specific approach.  
+      )
+
+    id :token_introspection_test_group
+
+    DEFAULT_INTR_BASE_URL = 'http://keycloak_auth_server:8080/realms/inferno'
+    DEFAULT_TOKEN_ENDPOINT = 'protocol/openid-connect/token'
+    DEFAULT_INTR_ENDPOINT = 'protocol/openid-connect/token/introspect'
+    DEFAULT_CLIENT_ID = 'inferno_client_confidential'
+    DEFAULT_CLIENT_SECRET = 'lLICFElPPfdcRQGnUcFjqAWexB1T6pqb'
```

Comment:
These could be extracted into a [preset](https://inferno-framework.github.io/inferno-core/deployment/input-presets.html).  Though perhaps it is premature as we would want to figure out this fits together with the existing tests / reference server.

Reply:
Thanks for the reference for presets.  I'm fine holding off on adding a preset for now - I think it would make sense once we narrow down our inputs and ref server approach like you said, and based on the ONC discussion yesterday the current ones will need to be revamped.  Let me know if you feel differently, for the time being I'll resolve the thread.